### PR TITLE
add DependentMaternGP to contrib.timeseries

### PR DESCRIPTION
**Pull request by martinjankowiak**
addresses #2090

### CODE REVIEW

File: pyro/contrib/timeseries/gp.py [hunk 1]
```diff
@@ -18,6 +20,12 @@ class IndependentMaternGP(TimeSeriesModel):
     :param float nu: The order of the Matern kernel; one of 0.5, 1.5 or 2.5.
     :param float dt: The time spacing between neighboring observations of the time series.
     :param int obs_dim: The dimension of the targets at each time step.
+    :param torch.Tensor log_length_scale_init: optional initial values for the kernel length scale
```
Comment (martinjankowiak) [inline pyro/contrib/timeseries/gp.py]: adding missing docstrings

File: pyro/contrib/timeseries/gp.py [hunk 2]
```diff
@@ -30,7 +38,7 @@ def __init__(self, nu=1.5, dt=1.0, obs_dim=1,
             log_obs_noise_scale_init = -2.0 * torch.ones(obs_dim)
         assert log_obs_noise_scale_init.shape == (obs_dim,)
 
-        super(IndependentMaternGP, self).__init__()
+        super().__init__()
```
Comment (martinjankowiak) [inline pyro/contrib/timeseries/gp.py]: simplify super() calls

File: pyro/contrib/timeseries/gp.py [hunk 3]
```diff
@@ -253,3 +267,159 @@ def forecast(self, targets, dts):
...
+
+        self.kernel = MaternKernel(nu=nu, num_gps=obs_dim,
+                                   log_length_scale_init=log_length_scale_init)
+        self.full_state_dim = self.kernel.state_dim * obs_dim
+
+        # we demote self.kernel.log_kernel_scale from being a nn.Parameter
+        # since the relevant scales are now encoded in the wiener noise matrix
+        del self.kernel.log_kernel_scale
+        self.kernel.register_buffer("log_kernel_scale", torch.zeros(obs_dim))
```
Comment (martinjankowiak) [inline pyro/contrib/timeseries/gp.py]: perhaps there's a better solution here?

Comment (fritzo) [reply pyro/contrib/timeseries/gp.py]: solution is good, but nit: might as well store `kernel_scale` if it is constant.

Comment (fritzo) [reply pyro/contrib/timeseries/gp.py]: do you want to fix this now, or should I merge?

**Comment by martinjankowiak** [comment]
I'll try to address that later,  thanks

**Pull request merged**